custom_model_runner/datarobot_drum/drum/drum.py

```diff
@@ -353,8 +359,68 @@ def run_test_predict(self):
                 df = handle_missing_colnames(df)
             df.to_csv(__tempfile.name, index=False)
             self.options.input = __tempfile.name
+        if self.target_type != TargetType.TRANSFORM:
+            self._check_prediction_side_effects()
         self._run_fit_and_predictions_pipelines_in_mlpiper()
 
+    def _check_prediction_side_effects(self):
```

Comment:
Do you mind working a bit on validation infrastructure:
* https://github.com/datarobot/datarobot-user-models/blob/master/custom_model_runner/datarobot_drum/drum/perf_testing.py - from here, move `validation_test` to a separate file
* move `_check_prediction_side_effects` to the new file/class created on the prev step.
* make validation check to perform these two validations
* in current place use `prediction side effects` validation as standalone

Reply:
can do

Reply:
...would you move the entire parent class of that method? I'm not super familiar with this corner of the codebase

Reply:
Lets start by moving `_check_prediction_side_effects` into existing `CMRunTests` class